libsolidity/codegen/ir/IRGeneratorForStatements.cpp

```diff
@@ -1576,19 +1576,6 @@ void IRGeneratorForStatements::endVisit(MemberAccess const& _memberAccess)
 
 	if (memberFunctionType && memberFunctionType->bound())
 	{
-		solAssert((set<Type::Category>{
-			Type::Category::Contract,
-			Type::Category::Bool,
-			Type::Category::Integer,
-			Type::Category::Address,
-			Type::Category::Function,
-			Type::Category::Struct,
-			Type::Category::Enum,
-			Type::Category::Mapping,
-			Type::Category::Array,
-			Type::Category::FixedBytes,
-		}).count(objectCategory) > 0, "");
```

Comment:
Wouldn't it be better to add the type `Type::Category::RationalNumber` and `StringLiteral` to the list?

Can the others be also be "bound"?

Reply:
I'm a bit conflicted about it. I can restore it if you think it is useful after all.

I think the assert was overly cautious because this should really handle every type in the same way:
```c++
define(IRVariable(_memberAccess).part("self"), _memberAccess.expression());
```

Back when I added it, I knew much less about IR implementation and I was not sure if the implementation would handle  anything except the types I managed to cover with test cases.

> Can the others be also be "bound"?

I think they can't but I can't be 100% sure. I did extensive testing back when I implemented it and the literals still managed to slip through.

For example there might be a way to bind array slices. Are slices implicitly convertible to array types? I did quick testing and it looks like they can't and I did not find any test cases for this. But if they can, that would be another case where the assert was wrong.